extensions/amp-story/0.1/media-pool.js

```diff
@@ -640,18 +597,20 @@ export class MediaPool {
    * already registered, this is a no-op.  Registering elements from within the
    * pool is not allowed, and will also be a no-op.
    * @param {!HTMLMediaElement} domMediaEl The media element to be registered.
+   * @return {!Promise} A promise that is resolved when the element has been
+   *     successfully registered, or rejected otherwise.
    */
   register(domMediaEl) {
     const mediaType = this.getMediaType_(domMediaEl);
     if (this.isAllocatedMediaElement_(mediaType, domMediaEl)) {
       // This media element originated from the media pool.
-      return;
+      return Promise.resolve();
     }
 
     const id = domMediaEl.id || this.createDomMediaElementId_();
```

Comment:
Looks like I'm missing something here, but should the result from ``this.createDomMediaElementId_()`` be assigned to ``domMediaEl.id``?

Reply:
Below, we set `domMediaEl.id = id`, which handles this.